ansible/playbook.yml

```diff
@@ -0,0 +1,65 @@
+---
+- name: Install tools
+  hosts: nodes
+  become: true
+  tasks:
+    - name: Install Git
+      yum:
+        name: git
+        state: present
+        update_cache: yes
+    - name: Install Python 3
+      yum:
+        name: python3
+        state: present
+        update_cache: yes
+
+- name: Install Poetry
+  hosts: nodes
+  tasks:
+    - name: Install Poetry
+      shell:
+        cmd: 'curl -sSL https://install.python-poetry.org | python3 -'
+        creates: ~/.local/bin/poetry
+
+- name: Create To-Do App Directory and clone from GitHub
+  hosts: nodes
+  become: true
```

Comment:
I don't know if this is the reason you've split the file out into several plays, but in case so worth noting that we can specify the `become` argument on a per-_task_ level, which would mean we could combine these all into a single play comfortably

Reply:
That said, it's very clear in your code that you've considered which steps need to be run as root and which don't, which is definitely a good thing!